### PR TITLE
Use -1 as a value for infinite number of writer attempts

### PR DESCRIPTION
**Pull request by VictorDenisov**

### CODE REVIEW

**Comment by VictorDenisov** [comment]
Fix for writer from ticket: https://github.com/segmentio/kafka-go/issues/243

**Comment by VictorDenisov** [comment]
@achille-roussel Could you please take a look at this PR. This fix is rather critical for me. Thanks in advance.

**Comment by VictorDenisov** [comment]
@stevevls maybe you can comment?

File: writer.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ type WriterConfig struct {
 	Balancer Balancer
 
 	// Limit on how many attempts will be made to deliver a message.
+	// -1 means that we can make unlimited number of attempts.
```
Comment (stevevls) [inline writer.go]: Could we make this a constant value instead?

**Comment by VictorDenisov** [comment]
Indeed MaxInt retries should be enough for any practical situation. Sorry for the disturbance. I'll close the PR.

**Pull request closed without merging**